contracts/transfer/src/state.rs

```diff
@@ -39,6 +41,24 @@ pub struct TransferState {
     var_crossover_pk: Option<PublicKey>,
 }
 
+impl TryFrom<Note> for TransferState {
+    type Error = Error;
+
+    /// This implementation is intended for test purposes to initialize the
+    /// state with the provided note
+    ///
+    /// To avoid abuse, the block_height will always be `0`
+    fn try_from(note: Note) -> Result<Self, Self::Error> {
+        let mut transfer = Self::new();
+
+        let block_height = 0;
+        transfer.push_note(block_height, note);
+        transfer.update_root();
+
+        Ok(transfer)
+    }
+}
```

Comment:
This definitely does not need to be here since we can't directly modify the state anyway - unless you want to use it in unit tests of the stake contract itself (which doesn't seem to be the case).

Reply:
It is used in TransferWrapper with_stakes method, I will leave it here until we refactor TransferWrapper